### DISCUSSION TITLE
Kong is Establishing too many UDP connections for dns resolution

### DISCUSSION THREAD

**Discussion by dsnr-coder** [Kong/kong#11136] [Help] [answered]
Hi All,

In Kong 3.1.1 version, kong is establishing too many UDP connections for dns resolution, is their way we can control the UDP connections.
![image](https://github.com/Kong/kong/assets/38465329/8db94329-4433-4cd2-8aee-e468b9946f6e)

![image](https://github.com/Kong/kong/assets/38465329/27eb3b32-8210-42ce-a521-6aa037fdd701)

Appreciate your help..

**Comment by ms2008** [comment, marked as answer]
Can you go try the latest version and see if the problem still exists? I remember we already fixed the DNS leak-related issue.

Ref: https://github.com/Kong/kong/pull/10691

**Reply by dsnr-coder** [reply]
thanks @ms2008 , I was going through kong github issues and corresponding fixes, I came across Ref: https://github.com/Kong/kong/pull/10691.
Currently we are upgrading to kong latest stable version 3.3.0, which has the above fix.

Thanks a lot for your help @ms2008
